Show actual sensor clippiing

jandren opened the pull request.

jandren commented on src/iop/rawoverexposed.c:
```diff
@@ -92,59 +92,21 @@ static void process_common_setup(dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *
-
     // "undo" rawprepare iop
     chthr *= piece->pipe->dsc.rawprepare.raw_white_point - piece->pipe->dsc.rawprepare.raw_black_level;
     chthr += piece->pipe->dsc.rawprepare.raw_black_level;
```
This seems like it could trigger a bug if the user or camera defines multiple different black levels. Maybe worth to use the actual separate black levels instead of their averages.

`pipe->dsc.rawprepare.raw_black_level` is the average of the `separate black levels`.

jandren replied:
Note that I meant lines 104 and 105 which are unchanged by this PR, i.e. not a regression.
It is an existing bug if we want to classify it as such but could be a good time to fix it while we are at it.

gi-man commented:
This PR sounds like a good idea to me. Is there an issue with merging it?

TurboGit commented:
> This PR sounds like a good idea to me. Is there an issue with merging it?

Yes, the dev says:

> This seems like it could trigger a bug if the user or camera defines multiple different black levels.

And I'm not yet sure this is correct or not. I'd like to have @aurelienpierre feedback on this.

jenshannoschwalm commented:
Have you tested the black marking to be fine with monochromes? :-)

jandren commented:
@jenshannoschwalm should equal false color in that use case, right? In regards to true black and white raws.

But yes, forgot about that usecase.
Another way of reducing the complexity of the UI but maintaining the current functionality would be to expose the solid color as a CSS option. It just seemed so excessive to have in there.

jandren commented:
Easy to revert to solid color change, so I'm waiting for the review until I do more changes to the PR.
Please add your opinions about simplifying the solid color selection until then.

jandren commented:
@TurboGit, @jenshannoschwalm, and @gi-man I simplified the PR to _only_ be the change to show actual raw sensor clipping (and rebased to the latest master).

I will leave the UI changes to a later follow-up PR.
Also did some better testing on the black level concern I had, seems to not be a problem/bug so marked that conversation as _solved_.

I.e. should be easier to review and merge now.

TurboGit commented:
> The current behavior takes the early "white balance normalization" into account. This is an old display referred assumption that relates to how the old "highlight reconstruction" method works.

And is that valid for people still using the display referred workflow (based on white balance)?

aurelienpierre commented:
I will have a look at that next week.

jandren commented:
> And is that valid for people still using the display referred workflow (based on white balance)?

It would not show the same as before no, but the tooltip says "Raw Overexposure" and I had liked to argue that there is only one way to define that. That way would be the actual raw sensor clipping, which is also independent of chosen workflow as it is tied to the camera and its sensor.

The old behavior is rather a visualization of where a module (highlights reconstruction) is active. That visualization could be readded as a "mask"/"visualize" option in the module. Especially as it has to be adopted based on the chosen reconstruction method when the new methods are added.

jenshannoschwalm commented:
> he tooltip says "Raw Overexposure" and I had liked to argue that there is only one way to define that

Agreed here.



> The old behavior is rather a visualization of where a module (highlights reconstruction) is active. That visualization could be readded as a "mask"/"visualize" option in the module. Especially as it has to be adopted based on the chosen reconstruction method when the new methods are added.

Agreed here too. BTW i have such a visualizing ready for highlights reconstruction module and could do a pr.

jandren commented:
Ping @aurelienpierre and @TurboGit

jandren commented:
Thanks!

@TurboGit I see that it already has been added to the release notes: https://github.com/darktable-org/darktable/commit/df9f404a4eb4b96254d8b7496f91a89b9eedd6ef

@elstoc Checking out the documentation https://darktable-org.github.io/dtdocs/en/module-reference/utility-modules/darkroom/raw-overexposed/ I'm inclined at saying that this update makes the software behave as described. So the documentation is already correct in my eyes :)

elstoc commented:
Thanks @jandren I'll take a look. I usually only add the pending tag as a reminder but I suspect you're probably right.
